File: apps/mobile/lib/models/canvassing_photo_policy.dart
bool prohibitsResidentialPhotos(Object? type) {
  final normalized = '$type'.toLowerCase().replaceAll(RegExp('[^a-z0-9]'), '');
  return const {
    'neighborhoodcanvassing',
    'canvassing',
    'flyerdistribution',
    'flyer',
    'doorhanger',
    'doorhangers',
    'doorhangerdistribution',
  }.contains(normalized);
}

File: functions/canvassing_photo_policy.js
"use strict";
// Server-owned job type, never a caller-selected photo exemption.
function prohibitsResidentialPhotos(campaign) {
  const value = String(campaign?.campaignType || campaign?.type || "")
    .toLowerCase().replace(/[^a-z0-9]/g, "");
  return new Set(["neighborhoodcanvassing", "canvassing", "flyerdistribution",
    "flyer", "doorhanger", "doorhangers", "doorhangerdistribution"]).has(value);
}
module.exports = {prohibitsResidentialPhotos};
